File: OCRP-Frontend/src/Services/CompanyService.js
import React,{Component} from "react";
import axios from "axios";

const CRS_URL = "http://localhost:9090/CRS_Backend/company";

class CompanyService extends Component {

  registerCompany(newCompany) {
    return axios.post(CRS_URL, newCompany);
  }

  getCompanyById(cId) {
      return axios.get(CRS_URL + "/" + cId);
  }

  updateCompanyById(cId,companyDetails) {
    return axios.put(CRS_URL + "/" + cId, companyDetails);
  }

  
  // getAllCompany() {
  //   return axios.get(CRS_URL);
  // }

  deleteCompanyById(cId) {
    return axios.delete(CRS_URL + "/" + cId);
  }

  postJob(cId,jobDetails) {
      return axios.post(CRS_URL + "/postjob" + cId, jobDetails);
  }

  updateJob(jId,jobDetails) {
    return axios.post(CRS_URL + "/updatejob" + jId, jobDetails);
}

  deleteJob(jId) {
    return axios.post(CRS_URL + "/deletejob" + jId);
}

}

export default new CompanyService();

File: OCRP-Backend/src/main/java/com/app/services/CompanyService.java
package com.app.services;

import java.util.List;

import org.springframework.http.ResponseEntity;

import com.app.pojos.Company;
import com.app.pojos.Job;

public interface CompanyService {

	public Company getCompanyById(int cId);

	public Company registerCompany(Company company);

	public String updateCompany(int cId, Company company);

	public String deleteCompanyById(int cId);

	public List<Company> getAllCompany();

	public String postJob(int cId,Job job);

	public String updateJob(int jId, Job job);

	public String deleteJobById(int jId);

	
}
